crypto/dh/dh_pmeth.c

```diff
@@ -422,7 +422,7 @@ static int pkey_dh_derive(EVP_PKEY_CTX *ctx, unsigned char *key,
             ret = DH_compute_key_padded(key, dhpubbn, dh);
         else
             ret = DH_compute_key(key, dhpubbn, dh);
-        if (ret < 0)
```

Comment:
Not against the change, but do note that this is a legacy code path.

The provider code is in providers/implementations/exchange/dh_exch.c
dh_plain_derive() does
````
    if (ret <= 0)
        return 0;
    ...
    return 1;
````

Reply:
Just to get it right: You're suggesting returning 0 from the error path? :)

Reply:
Just an observation that this is not the normal code path now..